src/Lib.hs

```diff
@@ -46,22 +48,51 @@ sha256 = digestFromByteString . hash . pack
 -- abstracted hash function that takes a string
 -- to hash and returns a hex string
 hashString :: String -> String
-hashString = maybe (error "Something went wrong generating a hash") show . sha256
+hashString =
+  maybe (error "Something went wrong generating a hash") show . sha256
 
 calculateBlockHash :: Block -> String
-calculateBlockHash (Block i p t b _)  = hashString $ concat [show i, p, show t, b]
+calculateBlockHash (Block i p t b n _)  =
+  hashString $ concat [show i, p, show t, b, show n]
 
 -- returns a copy of the block with the hash set
 addHashToBlock :: Block -> Block
 addHashToBlock block = block { blockHash = calculateBlockHash block }
 
+-- Rudimentary proof-of-work: ensures that a block hash
+-- is less than a certain value (i.e. contains a certain
+-- amount of leading zeroes)
+-- In our case, it's 4 leading zeroes. We're using the Integer type
+-- since the current target is higher than the max for Int
+difficultyTarget :: Integer
+difficultyTarget =
+  0x0000ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
+-- checks whether the provided block hash satisfies
+-- our PoW requirement
+satisfiesPow :: String -> Bool
+satisfiesPow blockHash = maybe
```

Comment:
this blockHash variable shadows the one exported from the Block type, so might be a good idea to rename to something else